packages/scripts/src/create-entrypoint.ts

```diff
@@ -365,10 +364,29 @@ function loadConfigFile(filePath: string, scopedName: string, configEnvName: str
 //#endregion
 
 //#region configs
+function populateConfig(envName: string, staticBuild?: boolean, publicConfigsRoute?: string, config?: TopLevelConfig) {
+    return `${staticBuild ? importStaticConfigs() : ''}
+${staticBuild && config ? addOverrideConfig(config) : ''}
+
+${publicConfigsRoute ? getRemoteConfigs(publicConfigsRoute, envName) : ''}
+
+${publicConfigsRoute ? `${addConfigsEventListenerForParentEnvironments(publicConfigsRoute)}` : ''}`;
```

Comment:
Switch the order of registering the listener and posting the message

Reply:
it doesn't matter in this case. only the parent environment registers the listener, and only the child env will access the parent using this listener

Reply:
Technically your right, in this case  it doesn't because everything is async, but as a practice it's a potential race.